app.js

```diff
@@ -88,10 +88,15 @@ async function GCMpush(certificate, device, payload) {
 		})
 		request.on('data', (chunk) => data.push(chunk))
 		request.on('end', () => {
+			try {
 			let response = JSON.parse(data.join())
 			if (response.success == 0)
 				 reject(response)
 			else resolve(response)
+			} catch (error) {
+			console.log("Error sending notifications - Android")
```

Comment:
@Linoy339 Can you remove the console log and instead return the error to the API caller (as done elsewhere in the codebase already)?

Reply:
@avaidyam 
Actually, this will return the error specified in [https://github.com/BIDMCDigitalPsychiatry/LAMP-app-gateway/blob/master/app.js#L149](https://github.com/BIDMCDigitalPsychiatry/LAMP-app-gateway/blob/master/app.js#L149.)

Reply:
@Linoy339 The issue is that you have not added a `throw error`, so while you have caught and logged an error, you are not rethrowing it to that function you linked to. Next, you're leaving the promise dangling as you've called neither `resolve` nor `reject`, and furthermore, we don't want `console.log`s at all in this codebase as the caller is the only that should know what errors occurred. 

**Please replace the `console.log(...)` with `reject(error)`.**

Reply:
Okay. Got it

Reply:
Please find[https://github.com/BIDMCDigitalPsychiatry/LAMP-app-gateway/pull/3]( https://github.com/BIDMCDigitalPsychiatry/LAMP-app-gateway/pull/3)